### PR TITLE
Split test file into smaller ones

### PR DESCRIPTION
**Pull request by sebaherrera07**
An attempt to split a large test file into smaller and more specific files.

I maintained the same existing tests and maybe just added a few when I noticed that would be good.

I'm open to any suggestions about the naming, grouping or any other thing about this proposal.

### CODE REVIEW

File: spec/ar2dto/record_to_dto/associations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          expect(subject.orders.first).to be_an(Shop::OrderDTO)
+          expect(subject.orders.first.user_id).to eq(user.id)
+          expect(subject.orders.first.user_id).to_not be_nil
+          expect(subject.person).to be_a(PersonDTO)
+          expect(subject.person.user_id).to eq(user.id)
+        end
+      end
+    end
+
+    context "when not including associations" do
```
Comment (sebaherrera07) [inline spec/ar2dto/record_to_dto/associations_spec.rb:57]: I added this test that didn't exist.

File: spec/ar2dto/relation_to_dto/attributes_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe ".to_dto" do
+  describe "attributes" do
```
Comment (sebaherrera07) [inline spec/ar2dto/relation_to_dto/attributes_spec.rb:6]: Same comment.

Not sure about the naming of this file and describe block.

I started by creating all the other files which seemed more specific, and the tests inside here were the "remaining" ones.

File: spec/ar2dto/relation_to_dto/associations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        expect(subject.first.person.user_id).to eq(user.id)
+        expect(subject.second.orders).to be_an(Array)
+        expect(subject.second.orders.first).to be_an(Shop::OrderDTO)
+        expect(subject.second.orders.first.user_id).to eq(another_user.id)
+        expect(subject.second.orders.first.user_id).to_not be_nil
+        expect(subject.second.person).to be_nil
+      end
+    end
+
+    context "when not including associations" do
```
Comment (sebaherrera07) [inline spec/ar2dto/relation_to_dto/associations_spec.rb:44]: I added this tests too.

File: spec/ar2dto/record_to_dto/attributes_spec.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+      it "doesn't expose ActiveRecord's methods" do
+        expect(subject).not_to respond_to(:update)
+      end
+
+      it "exposes methods to access the columns" do
+        expect(subject).to have_attributes(attributes)
+      end
+
+      it "exposes methods to access the columns set by persistance" do
```
Comment (martinjaimem) [inline spec/ar2dto/record_to_dto/attributes_spec.rb:58]:
```suggestion
      it "exposes methods to access the columns set by persistence" do
```

File: spec/ar2dto/options_spec.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "spec_helper"
```
[the remark this answers is not in the document]
Comment (santib) [reply spec/ar2dto/options_spec.rb:3]: isn't it needed if you want to run this file separately?

[the remark this answers is not in the document]
Comment (santib) [reply spec/ar2dto/options_spec.rb:3]: oh, I just tried and it worked 🤔 ahh it might be that it's here `.rspec`

**Review by martinjaimem — APPROVED** [review]

**Pull request merged**